shell/browser/native_window_mac.mm

```diff
@@ -1810,6 +1812,27 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   [window_ setAcceptsMouseMovedEvents:forward];
 }
 
+gfx::Rect NativeWindowMac::GetWindowControlsOverlayRect() {
+  gfx::Rect bounding_rect;
+  if (!has_frame() && buttons_view_ && ![buttons_view_ isHidden] &&
+      !traffic_light_position_) {
+    NSRect button_frame = [buttons_view_ frame];
+    const int overlay_width = GetContentSize().width() - NSWidth(button_frame);
+    int overlay_height = NSHeight(button_frame) + 6;
+    if (title_bar_style_ == TitleBarStyle::kHiddenInset) {
+      overlay_height += 15;
+    }
+    if (base::i18n::IsRTL()) {
+      bounding_rect = gfx::Rect(0, 0, overlay_width - button_frame.size.width,
+                                overlay_height);
+    } else {
+      bounding_rect =
+          gfx::Rect(button_frame.size.width, 0, overlay_width, overlay_height);
```

Comment:
The bounds calculation seems to be wrong, you can test with   https://amandabaker.github.io/pwa/windowControlsOverlay-newCSSVars/index.html.

Chrome Canary vs Electron:
<img width="1394" alt="Screen Shot 2021-06-01 at 17 11 45" src="https://user-images.githubusercontent.com/639601/120289728-8db61680-c2fc-11eb-90df-f6b97d7c3185.png">

Reply:
ok the bounds calculation should now be closer to the Chromium implementation but it will be slightly different in Electron due to the placement of the traffic lights.